src/libraries/System.Data.Common/src/System/Xml/XmlDataDocument.cs

```diff
@@ -2553,8 +2551,21 @@ public XmlDataDocument(DataSet dataset) : base(new XmlDataImplementation())
 
         internal XmlDataDocument(XmlImplementation imp) : base(imp)
         {
-        }
-
+            // This constructor is used by XmlDataImplementation.CreateDocument(), which
+            // exposes it as XmlDocument. The methods using these fields are never called.
```

Comment:
I'm not sure if this comment is 100% accurate. From looking at the code, it looks like `XmlDataImplementation.CreateDocument()` can be called publicly:

```C#
using System.Xml;

var x = new XmlDataDocument();
x.AppendChild(x.CreateElement("h"));

var y = (XmlDataDocument)x.Implementation.CreateDocument();

Console.WriteLine(y.DataSet?.ToString() ?? "<null>");
```

The code above prints `<null>`.

Reply:
That is fair. But a document constructed from there will also null ref, for example:

```C#
y.GetRowFromElement(y.DocumentElement);
```

Without some surgery I don't believe an `XmlDataDocument` that was created via `XmlDataImplementation.CreateDocument()` can be used safely today 😢